### PR TITLE
Add tests for Policy-Management and BGPVPN.

### PR DESCRIPTION
**Pull request by Goutham-Pratapa**
http://www.opencontrail.org/documentation/api/r5.0/contrail_openapi.html#policy-management-label
http://www.opencontrail.org/documentation/api/r5.0/contrail_openapi.html#bgpvpn-label

### CODE REVIEW

**Comment by aveeshek** [comment]
Can you provide the test results and pylint report without any pylint rc, please?

**Comment by Goutham-Pratapa** [comment]
https://sdp.web.att.com/fa3ccm1/web/projects/Integration_Infrastructure_LM_20.20#action=com.ibm.team.workitem.viewWorkItem&id=454677 

https://sdp.web.att.com/fa3ccm1/web/projects/Integration_Infrastructure_LM_20.20#action=com.ibm.team.workitem.viewWorkItem&id=454711

we already have them here right

**Comment by Goutham-Pratapa** [comment]
Here are the **Pylint-results** 

(35_env) root@ubuntu:~# pylint /home/ubuntu/run/tungsten-tempest/tungsten_tempest_plugin/tests/api/contrail/test_bgpvpn.py 

------------------------------------
Your code has been rated at 10.00/10

(35_env) root@ubuntu:~# pylint /home/ubuntu/run/tungsten-tempest/tungsten_tempest_plugin/tests/api/contrail/test_policy_management.py 

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

**Tempest-results:**

**admin**
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_create_policy_management [6.218197s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_delete_policy_management [6.669735s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_list_policy_managements [5.397828s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_show_policy_management [5.394721s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_update_policy_management [5.244792s] ... ok

======
Totals
======
Ran: 5 tests in 28.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 28.9253 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:28.941050

{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_create_bgpvpns [6.451468s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_delete_bgpvpn [6.440947s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_list_bgpvpns [5.528035s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_show_bgpvpn [5.527011s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_update_bgpvpn [5.305421s] ... ok

======
Totals
======
Ran: 5 tests in 29.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 29.5449 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:29.260100


**admin_contrail**

{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_create_bgpvpns [6.431468s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_delete_bgpvpn [6.450947s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_list_bgpvpns [5.527035s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_show_bgpvpn [5.528011s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_update_bgpvpn [5.307421s] ... ok

======
Totals
======
Ran: 5 tests in 29.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 29.2449 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:29.260100

{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_create_policy_management [6.454020s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_delete_policy_management [6.346875s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_list_policy_managements [5.635676s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_show_policy_management [5.508522s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_update_policy_management [5.494641s] ... ok

======
Totals
======
Ran: 5 tests in 29.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 29.4397 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:29.470934

**admin_support**
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_create_bgpvpns [6.351945s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_delete_bgpvpn [6.754163s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_list_bgpvpns [5.659587s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_show_bgpvpn [5.378873s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_update_bgpvpn [5.535969s] ... ok

======
Totals
======
Ran: 5 tests in 29.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 29.6805 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:29.695315

{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_create_policy_management [6.741101s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_delete_policy_management [6.580997s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_list_policy_managements [5.496216s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_show_policy_management [5.450331s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_update_policy_management [5.414610s] ... ok

======
Totals
======
Ran: 5 tests in 29.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 29.6833 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:29.706778

**_member_**

{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_create_bgpvpns [6.426032s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_delete_bgpvpn [6.841574s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_list_bgpvpns [5.449463s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_show_bgpvpn [5.703570s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_update_bgpvpn [5.036730s] ... ok

======
Totals
======
Ran: 5 tests in 29.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 29.4574 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:29.470092

{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_create_policy_management [6.459113s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_delete_policy_management [6.537438s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_list_policy_managements [5.541407s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_show_policy_management [5.613213s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_update_policy_management [5.642811s] ... ok

======
Totals
======
Ran: 5 tests in 29.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 29.7940 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:29.809052

**snapshot_member**
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_create_bgpvpns [6.451870s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_delete_bgpvpn [6.543311s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_list_bgpvpns [5.425310s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_show_bgpvpn [5.431354s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_update_bgpvpn [5.595536s] ... ok

======
Totals
======
Ran: 5 tests in 29.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 29.4474 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:29.477456

{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_create_policy_management [6.666415s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_delete_policy_management [8.187840s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_list_policy_managements [5.253190s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_show_policy_management [5.412573s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_update_policy_management [5.569651s] ... ok

======
Totals
======
Ran: 5 tests in 31.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 31.0897 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:31.103346

**Sriov_member**

{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_create_policy_management [6.477028s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_delete_policy_management [6.546091s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_list_policy_managements [5.215769s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_show_policy_management [5.502997s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_update_policy_management [5.359199s] ... ok

======
Totals
======
Ran: 5 tests in 29.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 29.1011 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:29.114304

{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_create_bgpvpns [7.889758s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_delete_bgpvpn [6.344267s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_list_bgpvpns [5.403473s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_show_bgpvpn [5.443681s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_update_bgpvpn [5.268566s] ... ok

======
Totals
======
Ran: 5 tests in 30.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 30.3497 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:30.368813

**support_member**

{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_create_bgpvpns [6.380822s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_delete_bgpvpn [6.660343s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_list_bgpvpns [5.437928s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_show_bgpvpn [5.236100s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_bgpvpn.BgpvpnTest.test_update_bgpvpn [5.391259s] ... ok

======
Totals
======
Ran: 5 tests in 29.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 29.1065 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:29.122146

{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_create_policy_management [6.942435s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_delete_policy_management [6.477204s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_list_policy_managements [5.431814s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_show_policy_management [5.424164s] ... ok
{0} tungsten_tempest_plugin.tests.api.contrail.test_policy_management.PolicyManagementTest.test_update_policy_management [5.318827s] ... ok

======
Totals
======
Ran: 5 tests in 29.0000 sec.
 - Passed: 5
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 29.5944 sec.

==============
Worker Balance
==============
 - Worker 0 (5 tests) => 0:00:29.611375

Thanks
Goutham Pratapa.

File: tungsten_tempest_plugin/tests/api/contrail/test_tag_type.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from patrole_tempest_plugin import rbac_rule_validation
+from tempest import config
+from tempest.lib.common.utils import data_utils
+from tempest.lib import decorators
```
Comment (SVilgelm) [inline tungsten_tempest_plugin/tests/api/contrail/test_tag_type.py]: ditto

**Pull request merged**